File: 0637-average-of-levels-in-binary-tree/0637-average-of-levels-in-binary-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def averageOfLevels(self, root: Optional[TreeNode]) -> List[float]:
        q = deque()
        q.append(root)
        res = []
        while q:
            l = len(q)
            s = 0
            for i in range(0, l):
                n = q.popleft()
                s += n.val
                if n.left:
                    q.append(n.left)
                if n.right:
                    q.append(n.right)
            res.append(s / l)
        return res
                

File: 0637-average-of-levels-in-binary-tree/0637-average-of-levels-in-binary-tree.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number[]}
 */
var averageOfLevels = function(root) {
    const res = []
    dfs(root, 0)
    return res.map(x => (x.reduce((acc, val) => acc + val, 0)) / x.length)
    
    function dfs(root, lvl) {
        if (!root) return
        if (!res[lvl]) res[lvl] = []
        res[lvl].push(root.val)
        dfs(root.left, lvl+1)
        dfs(root.right, lvl+1)
    }
};